Library Catalog (DRA).js

```diff
@@ -9,80 +9,195 @@
+		if (url.indexOf("/do_keyword_search/")!=-1 || url.indexOf("/form/")>-1) {
+			//Here we need first to go to another website which will then also
+			//create a session to continue.
+			if (marclink) {
+				url = "http://" + doc.location.host + marclink;
+			} else {
+				var pos = metalink.indexOf("URL=");
+				var link = metalink.substring(pos+4).replace('/log_out/', '/see_record/');
+				url = "//" + doc.location.host + link;
+			}
+			ZU.processDocuments([url], scrape);
+		} else {
+			scrape(doc, url);
 		}
-		scrape([url]);
 	}
 }
-function scrape(uris){
-	for(var i in uris) {
-		var uri = uris[i];
-		var uriRegexp = /^(https?:\/\/.*\/web2\/tramp2\.exe\/)(?:goto|see\_record|authority\_hits)(\/.*)\?(?:screen=Record\.html\&)?(.*)$/i;
-		var m = uriRegexp.exec(uri);
-		if(uri.indexOf("/authority_hits") < 0) {
-			var newUri = m[1]+"download_record"+m[2]+"/RECORD.MRC?format=marc&"+m[3];
-		} else {
-			var newUri = m[1]+"download_record"+m[2]+"/RECORD.MRC?format=marc";
-		}
-		
-		// Keep track of how many requests have been completed
-		var j = 0;
-		
-		var translator = Zotero.loadTranslator("import");
-		translator.setTranslator("a6ee60df-1ddc-4aae-bb25-45e0537be973");
-		
-		var domain = uri.match(/https?:\/\/([^/]+)/);
-		translator.setHandler("itemDone", function(obj, item) {
-			item.repository = domain[1]+" Library Catalog";
-			item.complete();
-		});
-		
-		Zotero.Utilities.HTTP.doGet(newUri, function(text) {
-			translator.setString(text);
-			translator.translate();		
-			j++;
-			if(j == uris.length) {
-				Zotero.done();
-			}
-		});
+
+
+function scrape(doc, uri) {
+	var uriRegexp = /^((?:https?:\/\/.*)?\/web2\/tramp2\.exe\/)(?:goto|see\_record|authority\_hits)(\/.*)\?(?:screen=Record\.html\&)?(.*)$/i;
+	var m = uriRegexp.exec(uri);
+	if(uri.indexOf("/authority_hits") < 0) {
+		var newUri = m[1]+"download_record"+m[2]+"/RECORD.MRC?format=marc&"+m[3];
+	} else {
+		var newUri = m[1]+"download_record"+m[2]+"/RECORD.MRC?format=marc";
+	}
+	
+	var translator = Zotero.loadTranslator("import");
+	translator.setTranslator("a6ee60df-1ddc-4aae-bb25-45e0537be973");
+	
+	var domain = uri.match(/https?:\/\/([^/]+)/);
+	translator.setHandler("itemDone", function(obj, item) {
+		item.repository = domain[1]+" Library Catalog";
+		item.complete();
+	});
+	
+	var encoding = "UTF-8";
+	//Sometimes the encoding is not UTF-8 but some multibyte extension of ASCII
+	//which uses besides the ASCII characters two bytes for encoding letters
+	//with diacritics. Before dealing with these we need to load the page as
+	//ISO-8859-1 encoded.
+	if (uri.indexOf("web2.libraries.vermont.gov/web2/tramp2.exe/")>-1 ||
+		uri.indexOf("catalogues.toulouse.fr/web2/tramp2.exe/")>-1) {
+		encoding = "ISO-8859-1";
+	}
+	
+	Zotero.Utilities.HTTP.doGet(newUri, function(text) {
+		text = encode(text);
+		translator.setString(text);
+		translator.translate();	
+	}, null, encoding);
+}
+
+
+function encode(text) {
```

Comment:
Well, the first problem is that I don't know whether the encoding has some more standard name. I tried for some time to find it out and I tried some other encodings, but in the end the only solution which I found was to write some mappings. It works...